tests/meta/upload/balsamic/test_balsamic.py

```diff
@@ -57,3 +70,97 @@ def test_genotype_check_only_tumour(balsamic_context: CGConfig):
+):
+
+    # GIVEN a Balsamic Upload API
+    upload_api = BalsamicUploadAPI(
+        create_autospec(
+            CGConfig,
+            delivery_path="delivery_path",
+            lims_api=Mock(),
+            balsamic=Mock(),
+            loqusdb=Mock(),
+            loqusdb_rd_lwp=Mock(),
+            loqusdb_wes=Mock(),
+            loqusdb_somatic=Mock(),
+            loqusdb_tumor=Mock(),
+            loqusdb_somatic_lymphoid=Mock(),
+            loqusdb_somatic_myeloid=Mock(),
+            loqusdb_somatic_exome=Mock(),
+            run_instruments=create_autospec(
+                RunInstruments,
+                illumina=create_autospec(IlluminaConfig, demultiplexed_runs_dir="runs_dir"),
+            ),
+            sentieon_licence_server=Mock(),
+            status_db=Mock(),
+        ),
+    )
+
+    # GIVEN a case containing a sample with a TGS prep category
+    case: Case = create_autospec(Case, internal_id="case_id")
+
+    cli_mock = mocker.patch.object(click.Context, "invoke")
+    mocker.patch.object(DeliverFilesService, "deliver_files_for_case")
+    mocker.patch.object(AnalysisAPI, "get_case_application_type", return_value=prep_category)
+
+    # WHEN uploading
+    upload_api.upload(ctx=click.Context(command=Mock()), case=case, restart=False)
+
+    # THEN it should upload observations to LoqusDB
+    cli_mock.assert_called_with(upload_observations_to_loqusdb, case_id=case.internal_id)
+    assert cli_mock.call_count == 2
+
+
+def test_loqusdb_upload_based_on_prep_category_fail(mocker: MockerFixture):
+    # GIVEN a Balsamic Upload API
+    upload_api = BalsamicUploadAPI(
+        create_autospec(
+            CGConfig,
+            delivery_path="delivery_path",
+            lims_api=Mock(),
+            balsamic=Mock(),
+            loqusdb=Mock(),
+            loqusdb_rd_lwp=Mock(),
+            loqusdb_wes=Mock(),
+            loqusdb_somatic=Mock(),
+            loqusdb_tumor=Mock(),
+            loqusdb_somatic_lymphoid=Mock(),
+            loqusdb_somatic_myeloid=Mock(),
+            loqusdb_somatic_exome=Mock(),
+            run_instruments=create_autospec(
+                RunInstruments,
+                illumina=create_autospec(IlluminaConfig, demultiplexed_runs_dir="runs_dir"),
+            ),
+            sentieon_licence_server=Mock(),
+            status_db=Mock(),
+        ),
+    )
+
+    # GIVEN a case containing a sample with a TGS prep category
+    case: Case = create_autospec(Case, internal_id="case_id")
+
+    cli_mock = mocker.patch.object(click.Context, "invoke")
+    mocker.patch.object(DeliverFilesService, "deliver_files_for_case")
+    mocker.patch.object(
+        AnalysisAPI,
+        "get_case_application_type",
+        return_value=SeqLibraryPrepCategory.WHOLE_TRANSCRIPTOME_SEQUENCING,
+    )
+
+    # WHEN uploading
+    upload_api.upload(ctx=click.Context(command=Mock()), case=case, restart=False)
+
+    # THEN it should upload observations to LoqusDB
+    assert cli_mock.call_count == 1
```

Comment:
This feels a bit off. We want to get input on whether someone can think of an alternative.

Reply:
removed test